ext/src/ruby_api/func.rs

```diff
@@ -116,11 +121,12 @@ pub fn make_func_closure(
     let ty = ty.to_owned();
     let callable = ShareableProc(callable);
 
-    move |caller: CallerImpl<'_, StoreData>, params: &[Val], results: &mut [Val]| {
-        let caller = RefCell::new(caller);
+    move |caller_impl: CallerImpl<'_, StoreData>, params: &[Val], results: &mut [Val]| {
+        let caller_value: Value = Caller::new(caller_impl).into();
+        let caller = caller_value.try_convert::<&Caller>().unwrap();
 
         let rparams = RArray::with_capacity(params.len() + 1);
-        rparams.push(Caller { inner: &caller }).ok();
+        rparams.push(caller_value).ok();
```

Comment:
I'll note here but applies in some other places in the codebase. Why are we not expecting or unwrapping the result of the `push`?

Reply:
Oops yeah, makes sense -- `unwrap` is what we want.

The only way this can fail is if the `Array` is frozen, which can't really happen: we just created the Array above.